File: backend-old/TransportStatic/Models/StopTime.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;

namespace TransportStatic.Models;

[Table("stop_times")]
public class StopTime
{
    [Column("trip_id")]
    public string TripId { get; set; } = null!;

    [Column("arrival_time")]
    public int ArrivalTime { get; set; }

    [Column("departure_time")]
    public int DepartureTime { get; set; }

    [Column("stop_id")]
    public string StopId { get; set; } = null!;

    [Column("stop_sequence")]
    public int StopSequence { get; set; }

    [Column("stop_headsign")]
    public string StopHeadSign { get; set; } = null!;

    [Column("pickup_type")]
    public int PickupType { get; set; }

    [Column("drop_off_type")]
    public int DropOffType { get; set; }

    [Column("shape_dist_travelled")]
    public decimal? ShapeDistanceTravelled { get; set; }

    [Column("timepoint")]
    public int? Timepoint { get; set; }

    [Column("stop_note")]
    public string? StopNote { get; set; }
    
    [Column("mode")]
    public string Mode { get; set; } = null!;
}

File: frontend/src/lib/types/stop-time.ts
export interface StaticStopTime {
  tripId: string
  stopId: string
  stopName: string
  arrivalTime: number
  departureTime: number
  stopSequence: number
  stopHeadsign: string
  pickupType: number
  dropOffType: number
  shapeDistTravelled: number
  timepoint: number
  stopNote: string
  mode: string
}

export interface RealtimeStopTime {
  stopId: string
  stopName: string
  stopPtSequence: number
  vehiclePtSequence: number
  stopSequence: number
  stopHeadsign: string
  arrivalTime: number
  departureTime: number
  arrivalDelay: number
  departureDelay: number
  status: string
  progress: string
  consist: Consist[]
}

export interface Consist {
  positionInConsist: number
  occupancyStatus: string
}